### PR TITLE
Fix `ResampleWithDistributionTransform` to work with integer timestamp

### LINKED ISSUE #160: Fix resample transform to handle integer timestamp
### 🚀 Feature Request

In #148 it was found that resample transforms fail on data with integer timestamp. It could be fixed.

### Proposal

Fix tests `tests/test_transforms/test_inference/test_transform/TestTransformTrain` for transforms:
- `DensityOutliersTransform`
- `MedianOutliersTransform`
- `PredictionIntervalOutliersTransform`

As I can see, the problem is within `_OneSegmentResampleWithDistributionTransform._get_folds`. It tries to infer freq. There are few possible ways to solve this. We should probably write a custom logic for handling integer timestamp here.

### Test cases

- Move fixed tests in `tests/test_transforms/test_inference/test_transform/TestTransformTrain` from fail to success category
- Move fixed tests `tests/test_transforms/test_inference/test_transform/TestInverseTransformTrain` from fail to success category

### Additional context

Blocked by #148.

### PR DESCRIPTION
**Pull request by d-a-bunin**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look #160.

## Closing issues
Closes #160.

### CODE REVIEW

File: etna/transforms/missing_values/resample.py [hunk 1]
```diff
@@ -111,6 +117,15 @@ def inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
 class ResampleWithDistributionTransform(IrreversiblePerSegmentWrapper):
     """ResampleWithDistributionTransform resamples the given column using the distribution of the other column.
 
+    This transform expects ``in_column`` to have non-NaN values separated by the same number of timestamps to form a cycle.
+    The cycle starts with a non-NaN value and each position has a number from 0 to cycle size - 1.
```
Comment (alex-hse-repository) [inline etna/transforms/missing_values/resample.py:121]: cycle size - 1 -> cycle_size - 1

File: etna/transforms/missing_values/resample.py [hunk 2]
```diff
@@ -111,6 +117,15 @@ def inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
...
+
+    During ``fit'', the fraction of each cycle position in a total sum of values is calculated according to ``distribution_column''.
```
Comment (alex-hse-repository) [inline etna/transforms/missing_values/resample.py:123]: 👍

File: etna/transforms/missing_values/resample.py [hunk 3]
```diff
@@ -44,13 +44,19 @@ def _get_folds(self, df: pd.DataFrame) -> List[int]:
...
         in_column_index = df[self.in_column].dropna().index
-        if len(in_column_index) <= 1 or (len(in_column_index) >= 3 and not pd.infer_freq(in_column_index)):
+
+        fail_datetime_timestamp = (
+            in_column_index.dtype != "int" and len(in_column_index) >= 3 and not pd.infer_freq(in_column_index)
+        )
+        if len(in_column_index) <= 1 or fail_datetime_timestamp:
             raise ValueError(
-                "Can not infer in_column frequency!"
+                "Can not infer in_column frequency! "
```
Comment (alex-hse-repository) [inline etna/transforms/missing_values/resample.py:53]: ?

File: etna/transforms/missing_values/resample.py [hunk 4]
```diff
@@ -44,13 +44,19 @@ def _get_folds(self, df: pd.DataFrame) -> List[int]:
         Here the ``in_column`` frequency gap is divided into the folds with the size of dataset frequency gap.
         """
         in_column_index = df[self.in_column].dropna().index
-        if len(in_column_index) <= 1 or (len(in_column_index) >= 3 and not pd.infer_freq(in_column_index)):
+
+        fail_datetime_timestamp = (
```
Comment (alex-hse-repository) [inline etna/transforms/missing_values/resample.py:48]: Looks like we can rewrite it in more general way (what if we have the other type of index?) and convert to some utils function which will infer freq from dataframe -- we won't do it here, just idea for the future

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]
As I understand we don't actually check the result, we have only some kind of smoke tests?

File: etna/transforms/missing_values/resample.py (hunk 4 above)
Comment (d-a-bunin) [reply etna/transforms/missing_values/resample.py:48]: Ok, may be we'll do smth like this in the future.

File: etna/transforms/missing_values/resample.py (hunk 3 above)
Comment (d-a-bunin) [reply etna/transforms/missing_values/resample.py:53]: I think we should add a space. Otherwise messages collapse together and it doesn't look nice.

**Review by alex-hse-repository — APPROVED** [review]

**Pull request merged**